activestorage/test/models/attached/many_test.rb

```diff
@@ -846,19 +846,35 @@ def highlights
     end
   end
 
-  test "attaching blobs to a record returns the attachments" do
-    @user.highlights.attach create_blob(filename: "racecar.jpg")
-    highlights = @user.highlights.attach create_blob(filename: "funky.jpg"), create_blob(filename: "town.jpg")
-    assert_instance_of ActiveStorage::Attached::Many, highlights
-    assert_equal 3, @user.highlights.count
-    assert_equal 3, highlights.count
-    assert_equal highlights.name.to_s, @user.highlights.name.to_s
-    assert_equal highlights.first.key.to_s, @user.highlights.first.key.to_s
-    assert_equal highlights.first.filename.to_s, @user.highlights.first.filename.to_s
-    assert_equal highlights.second.key.to_s, @user.highlights.second.key.to_s
-    assert_equal highlights.second.filename.to_s, @user.highlights.second.filename.to_s
-    assert_equal highlights.third.key.to_s, @user.highlights.third.key.to_s
-    assert_equal highlights.third.filename.to_s, @user.highlights.third.filename.to_s
+  test "attaching blobs to a persisted, unchanged, and valid record, returns the attachments" do
+    return_value = @user.highlights.attach create_blob(filename: "funky.jpg"), create_blob(filename: "town.jpg")
+    assert_instance_of ActiveStorage::Attached::Many, return_value
+    assert_equal @user.highlights, return_value
+  end
+
+  test "attaching blobs to a persisted, unchanged, and invalid record, returns nil but still attaches" do
+    @user.update_attribute(:name, nil)
+    assert_not @user.valid?
+
+    return_value = @user.highlights.attach create_blob(filename: "funky.jpg"), create_blob(filename: "town.jpg")
+    assert_nil return_value
+    assert_instance_of ActiveStorage::Attached::Many, @user.highlights
+    assert ActiveStorage::Blob.service.exist?(@user.highlights.first.key)
+    assert ActiveStorage::Blob.service.exist?(@user.highlights.second.key)
+  end
```

Comment:
@jonathanhefner look at this case. It's very strange that we are returning `nil`.. because the attachment is still created: the blob, and even the upload to the service 👀 Am I missing something?

Reply:
ok, I understand it now. The attachment is kept in memory, so if you save the record later it will be attached, if you reload you lose it. I wouldn't say returning nil but keeping it in memory is the most intuitive behavior but I can't think of anything better.